2.5-huffman.ipynb

```diff
@@ -1,25 +1,248 @@
 {
  "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import util\n",
+    "from test_framework.key import ECKey, ECPubKey, generate_key_pair\n",
+    "from test_framework.script import TapTree, TapLeaf, Node"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# 2.5 Huffman TapTree Constructor\n",
+    "\n",
+    "When constructing a TapTree, there is the possibility of optimizing the tree structure so that the expected witness weight when spending along the script path is minimized. The `expected witness weight` is determined by summing up all probability-adjusted witness weights of the TapScripts in the TapTree.\n",
+    "\n",
+    "For example, for a TapTree with TapLeafs A, B, and C, the expected witness weight is determined as follows:\n",
+    "\n",
+    "* **`Expected-witness-weight =`**\n",
```

Comment:
I think this looks better without the bullet points, for example:

```
Expected-witness-weight =
      Probability-of-A * Witness-weight-A
    + Probability-of-B * Witness-weight-B
    + Probability-of-C * Witness-weight-C
```

(ie use the backticks in your markdown to get fixed-width rendering)

Reply:
Thanks for this, this is better. Fixed.